frontend/components/forms/fields/Dropdown/Dropdown.jsx

```diff
@@ -56,14 +58,26 @@ class Dropdown extends Component {
   };
 
   handleChange = (selected) => {
-    const { multi, onChange, clearable } = this.props;
-
-    if (clearable && selected === null) {
-      onChange(null);
-    } else if (multi) {
-      onChange(selected.map((obj) => obj.value).join(","));
+    const { multi, onChange, clearable, name, target } = this.props;
+
+    console.log("onchange dropdown!");
+    console.log("target", target);
+    console.log("name", name);
+    console.log("selected.value", selected.value);
+    if (target) {
+      if (clearable && selected === null) {
+        onChange(name, null);
+      } else {
+        onChange(name, selected.value);
+      }
     } else {
-      onChange(selected.value);
+      if (clearable && selected === null) {
+        onChange(null);
+      } else if (multi) {
+        onChange(selected.map((obj) => obj.value).join(","));
+      } else {
+        onChange(selected.value);
+      }
```

Comment:
This block is getting a little if-fy. One trick is using `return` to short circuit the function like this: 

```javascript
    if (target) {
      clearable && selected === null
        ? onChange(name, null)
        : onChange(name, selected.value);
      return;
    }

    if (clearable && selected === null) {
      onChange(null);
      return;
    }

    if (multi) {
      onChange(selected.map((obj) => obj.value).join(","));
      return;
    }

    onChange(selected.value);
```